### PR TITLE
metadata and dashboard api

### PR DESCRIPTION
**Pull request by shrikantdange1208**

### CODE REVIEW

File: functions/api/dashboard.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+/**
+ * @description Route to retireve all data for the dashboard from firestore
+ * If the user is Admin or SuperAdmin return all dashboard data from all Branches
+ * If the user is Branch Userreturn dashboard data from that specific branch
+ * @returns Json object containing all metadata
+ */
+router.get("/", async (request, response, next) => {
+
+    const userRole = request.user.role
```
Comment (neerk28) [inline functions/api/dashboard.js:16]: can you add a validation here to check if role is one of the three roles

File: functions/api/dashboard.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    console.debug(`Retrieving Dashboard data for ${userRole.toUpperCase()}`)
+    if (userRole === constants.ADMIN || userRole === constants.SUPER_ADMIN) {
+
+        //Get dashboard data for all branches
+        const dashboardDataForAllBranches = {}
+        const dashboardData = await getDashboardDataForAdminUser()
+        dashboardData.forEach(dashboard => {
+            const branchId = dashboard[constants.BRANCH]
+            delete dashboard[constants.BRANCH]
```
Comment (neerk28) [inline functions/api/dashboard.js:25]: why are we deleting the branch?

File: functions/api/dashboard.js (hunk 1 above)
Comment (neerk28) [inline functions/api/dashboard.js:16]: i thought vinayak wanted role as query param. Admin can see entire view or just branch view by passing role?

File: functions/api/roles.js [hunk 3]
```diff
@@ -46,6 +46,14 @@ const { isAdminOrSuperAdmin } = require('../middleware/auth')
 2.SuperAdmin,Admin,Branch are the roles supported currently.
 */
 router.get('/', isAdminOrSuperAdmin, async (req, res, next) => {
```
Comment (neerk28) [inline functions/api/roles.js:48]: q: do we need isAdminOrSuperAdmin check for get roles? do we show roles only to admin?

File: functions/api/dashboard.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            dashboardDataForAllBranches[branchId] = dashboard
+        })
+        console.debug('Retrieved dashboard data for all the branches')    
+        response.status(200).json(dashboardDataForAllBranches)
+
+    } else if (userRole === constants.BRANCH) {
+        
+        // Get dashboard data for branch 
+        const branchId = request.user.branch
```
Comment (neerk28) [inline functions/api/dashboard.js:34]: if admin wants to view specific branch he cant right? not sure if you guys discussed in last call.

**Review by neerk28 — CHANGES_REQUESTED** [review]
Have some clarifications. Also can you add metadata and dashboard.json with sample response,For us to know the response structure

File: functions/api/dashboard.js (hunk 2 above)
Comment (shrikantdange1208) [reply functions/api/dashboard.js:25]: I have reused the old methods but Vinayak wants it in different format. So, I had to delete the branch.

File: functions/api/dashboard.js (hunk 1 above)
Comment (shrikantdange1208) [reply functions/api/dashboard.js:16]: No, this role is set after the user is authenticated in the auth, right? I am just reusing that.

File: functions/api/roles.js (hunk 3 above)
Comment (shrikantdange1208) [reply functions/api/roles.js:48]: Now it's metadata. They will have to read all top level collections. So we need to check only if the user is authenticated.
Also, the Read routes which we have written with different filters doesn't make sense :)

File: functions/api/dashboard.js (hunk 4 above)
Comment (shrikantdange1208) [reply functions/api/dashboard.js:34]: He can.

**Review by ramyarv — APPROVED** [review]

**Pull request merged**